support/metrics_plaid.py

```diff
@@ -189,329 +35,79 @@ def plaid_kyc(acc, txn):
-        feedback (dict): feedback describing the score
-    '''
-    try:
-        id = dynamic_select(acc, txn, 'credit', feedback)['id']
-        alltxn = [t for t in txn if t['account_id'] == id]
-
-        if alltxn:
-            oldest_txn = alltxn[-1]['date']
-            # date today - date of oldest credit transaction
-            how_long = (NOW - oldest_txn).days
-            score = params['fico_medians'][
-                np.digitize(how_long, params['duration'], right=True)
-            ]
-
-            feedback['credit']['credit_duration_days'] = how_long
-
-        else:
-            raise Exception('no credit length')
-
-    except Exception as e:
-        score = 0
-        feedback['credit']['error'] = str(e)
-
-    finally:
-        return score, feedback
-
-
-# @evaluate_function
-def credit_livelihood(acc, txn, feedback, params):
-    '''
-    Description:
-        returns score quantifying the avg monthly txn count for your best credit account
-
-    Parameters:
-        acc (list): Plaid 'Accounts' product
-        txn (list): Plaid 'Transactions' product
-        feedback (dict): score feedback
-        params (dict): model parameters, i.e. coefficients
-
-    Returns:
-        score (float): based on avg monthly txn count
-        feedback (dict): feedback describing the score
+    score[list] order must be the same as showed under metrics in the cofig.json file, e.g.
+    "data": [{"minimum_requirements": {"plaid": {"scores": {"models": {"credit": {"metrics": {...}}}}}}}]
     '''
+    score = []
     try:
-        id = dynamic_select(acc, txn, 'credit', feedback)['id']
-        alltxn = [t for t in txn if t['account_id'] == id]
-
-        if alltxn:
-            dates = list()
-            amounts = list()
-
-            for i in range(len(alltxn)):
-                dates.append(alltxn[i]['date'])
-                amounts.append(alltxn[i]['amount'])
-
-            df = pd.DataFrame(data={'amounts': amounts}, index=pd.DatetimeIndex(dates))
-            d = df.groupby(pd.Grouper(freq='M')).count()
-            credit_livelihood.d = d
-
-            if len(d['amounts']) >= 2:
-                if d['amounts'][0] < 5:  # exclude initial and final month with < 5 txn
-                    d = d[1:]
-                if d['amounts'][-1] < 5:
-                    d = d[:-1]
-
-            mean = d['amounts'].mean()
-            score = params['fico_medians'][
-                np.digitize(mean, params['count_lively'], right=True)
-            ]
-
-            feedback['credit']['avg_count_monthly_txn'] = round(mean, 0)
+        if metadata['credit_card']:
+            # read metadata
+            d = metadata['credit_card']['general']
+            acc_count = d['accounts']['total_count']
+            txn_avg_count = d['transactions']['avg_monthly_count']
+            txn_timespan = d['transactions']['timespan']
+            bal_limit = sum(d['balances']['limit'])  # shouldnt be average?
```

Comment:
- No, it **shouldn't** be an average. This metric is looking at how much money (in total) have available monthly. That quantity is given by the **sum** (not the average) a user's credit limit across all credit cards.